---
title: "Lamprey RNA-seq analysis"
author: "Megan Rothstein"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(biomaRt)
library(tidyverse)
library(DESeq2)
library(ggrepel)
library(pheatmap)
library(edgeR)
library(RColorBrewer)
```


Load data
```{r}
kpetMar=read.table("/data/Megan/Lamprey_CT/kpetMar1.featurecounts.txt", header=T)
orths = read.table("/data/Megan/Lamprey_CT/Lamprey_bt2/Ortholog_gene_models.txt", sep="\t", header=T)
```

make metadata table
```{r}
rownames(kpetMar) = kpetMar$Geneid
kpetMar = kpetMar[,7:10]
kpetMar = kpetMar %>% dplyr::rename('Cranial_R1'='..bam.PmCranial_R1_kpetMar1.bam', 
                          'Cranial_R2'='..bam.PmCranial_R2_kpetMar1.bam',
                          'Trunk_R1'='..bam.PmTrunk_R1_kpetMar1.bam',
                          'Trunk_R2'='..bam.PmTrunk_R2_kpetMar1.bam')


DESeq_design = data.frame(row.names = colnames(kpetMar),
           condition = c("C","C","T","T"),
           libType = c("single","single","single","single"))
DESeq_design
```

Make DESeq dataset
```{r}
kpetMar_dds <- DESeqDataSetFromMatrix(countData = kpetMar,
                              colData = DESeq_design,
                              design = ~condition)
kpetMar_dds$condition <- factor(kpetMar_dds$condition, levels = c("T","C"))
kpetMar_dds
```


Run DESeq
```{r}
kpetMar_dds <- DESeq(kpetMar_dds)
kpetMar_res <- results(kpetMar_dds)
kpetMar_resDf <- as.data.frame(kpetMar_res)
kpetMar_resDf
```


```{r}
kpetMar_resDf$PMGeneID = rownames(kpetMar_resDf)
kpetMar_resDf = left_join(kpetMar_resDf, orths, by='PMGeneID')
kpetMar_resDf
```


Node genes volcano plot
```{r}
x= list("GSC","CHRD","DKK1","CNOT2","OTX2","FOXA2","LHX1","EOMES","HHEX","SFRP1","SHH",
        "CRHR1","ADMP","FZD5","LBH","SPRY1","NODAL","GATA4","LMX1A","SMAD9","GMDS",
        "SPATA5","MAP4K4","TRABD2B","SH3PXD2A","LDLRAD3","PLEKHM3","P4HTM","LPP")

kpetMar_volcano = ggplot(kpetMar_resDf, aes(x=log2FoldChange, y=-log10(pvalue))) + 
  theme_classic() +
  geom_point(color = "gray", alpha = 0.5) +
  geom_point(data = filter(kpetMar_resDf, GeneName %in% x), color="#2177f6", size=2.5) +
  geom_text_repel(data = filter(kpetMar_resDf, GeneName %in% x), 
                  aes(label=GeneName), box.padding = 0.75, max.overlaps=100)
kpetMar_volcano
ggsave("kpetMar_volcano.pdf", width=4.5, height = 4)
```



