
SAMPLES = ['ggcranial_rep1','ggcranial_rep2','ggcranial_rep3','ggtrunk_rep1','ggtrunk_rep2','ggtrunk_rep3']

rule all:
	input:
		expand("fastq/{sample}_fastqc.html", sample=SAMPLES),
		expand("trimmed_fastq/trimmed_{sample}.fastq.gz", sample=SAMPLES),
		expand("bam/galGal6/{sample}_galGal6.bam", sample=SAMPLES),
		expand("bam/galGal6/{sample}_galGal6.bam.bai", sample=SAMPLES)

rule fastqc:
	input:
		"fastq/{sample}.fastq.gz"
	output:
		"fastq/{sample}_fastqc.html"
	threads: 8
	shell:
		"fastqc {input} -t {threads}"


rule cutadapt:
	input:
		"fastq/{sample}.fastq.gz"
	output:
		"trimmed_fastq/trimmed_{sample}.fastq.gz"
	log:
		"logs/cutadapt/{sample}.log"
	threads: 8
	shell:
		"cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC \
		--minimum-length=25 \
		--cores={threads} \
		{input} \
		-o {output} 2> {log}"


rule hisat2:
	input:
		"trimmed_fastq/trimmed_{sample}.fastq.gz"
	output:
		bam="bam/galGal6/{sample}_galGal6.bam"
	log:
		"logs/bowtie2/{sample}_galGal6.log"
	threads: 16
	shell:
		"hisat2 \
    	-p {threads} \
    	--phred33 \
    	--no-unal \
    	-x /data/Megan/genome_data/galGal6/galGal6_UCSC_HT2/galGal6_ht2_UCSC \
        -U {input} \
        2> {log} | samtools view -bh - | samtools sort -T {wildcards.sample}_galGal6 - -o {output.bam}"


rule index:
	input:
		"bam/galGal6/{sample}_galGal6.bam"
	output:
		"bam/galGal6/{sample}_galGal6.bam.bai"
	shell:
		"samtools index {input}"


rule featurecounts:
	input:
		"bam/galGal6/{sample}_galGal6.bam"
	output:
		"counts/hiNCC_featureCounts.txt"
	threads: 8
	shell:
		"featureCounts \
		{input} \
		-T {threads} \
		-t exon \
		-s 0 \
		-g gene_name \
		-a /data/Megan/genome_data/galGal6/galGal6.ensGene.gtf \
		-o {output}"




